yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/DataEntryConverterTest.java

```diff
@@ -0,0 +1,82 @@
+package com.yoti.api.client.spi.remote;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.yoti.api.client.ExtraDataException;
+import com.yoti.api.client.credentials.CredentialIssuanceDetails;
+import com.yoti.api.client.spi.remote.credentials.ThirdPartyAttributeConverter;
+import com.yoti.api.client.spi.remote.proto.DataEntryProto;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.text.ParseException;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DataEntryConverterTest {
+
+    private static final String SOME_DATA_ENTRY = "someBytes";
+    private static final ByteString SOME_DATA_ENTRY_BYTE_STRING = ByteString.copyFromUtf8(SOME_DATA_ENTRY);
+
+    @InjectMocks
+    DataEntryConverter dataEntryConverter;
+
+    @Mock
+    ThirdPartyAttributeConverter thirdPartyAttributeConverterMock;
+    @Mock
+    DataEntryProto.DataEntry dataEntryMock;
+    @Mock
+    CredentialIssuanceDetails credentialIssuanceDetailsMock;
+
+    @Test
+    public void shouldParseDataEntryCorrectly() throws Exception {
+        when(dataEntryMock.getType()).thenReturn(DataEntryProto.DataEntry.Type.THIRD_PARTY_ATTRIBUTE);
+        when(dataEntryMock.getValue()).thenReturn(SOME_DATA_ENTRY_BYTE_STRING);
+        when(thirdPartyAttributeConverterMock.parseThirdPartyAttribute(SOME_DATA_ENTRY_BYTE_STRING.toByteArray())).thenReturn(credentialIssuanceDetailsMock);
+
+        dataEntryConverter.convertDataEntry(dataEntryMock);
+    }
+
+    @Test
+    public void shouldThrowExceptionForUnsupportedProtobufType() {
+        when(dataEntryMock.getType()).thenReturn(DataEntryProto.DataEntry.Type.UNDEFINED);
+
+        try {
+            dataEntryConverter.convertDataEntry(dataEntryMock);
+        } catch (ExtraDataException e) {
+            assertThat(e.getMessage(), containsString("Unsupported data entry"));
+            return;
+        }
+        fail("Expected ExtraDataException");
+    }
+
+    @Test(expected = ExtraDataException.class)
+    public void shouldWrapInvalidProtocolBufferException() throws Exception {
+        InvalidProtocolBufferException invalidProtocolBufferException = new InvalidProtocolBufferException("Some Exception");
+
+        when(dataEntryMock.getType()).thenReturn(DataEntryProto.DataEntry.Type.THIRD_PARTY_ATTRIBUTE);
+        when(dataEntryMock.getValue()).thenReturn(SOME_DATA_ENTRY_BYTE_STRING);
+        when(thirdPartyAttributeConverterMock.parseThirdPartyAttribute(SOME_DATA_ENTRY_BYTE_STRING.toByteArray())).thenThrow(invalidProtocolBufferException);
+
+        dataEntryConverter.convertDataEntry(dataEntryMock);
+    }
```

Comment:
Should there be any assertions here? Should we be asserting that the exception is of the expected type?

Reply:
No - we're just checking that it managed to create the data entry successfully and didn't throw any exceptions.  The method `throws Exception` so that the test will fail if any exceptions are thrown.

Reply:
This test is just to make sure that the exception is wrapped into an `ExtraDataException`.  It's more generic than the one you have described above to make sure that any exceptions of a certain type are wrapped, not in a specific place which is what you have described does.  We can add in some more specific tests though 👍

Reply:
Ah ok, that's fine, I was just worried that it was just checking for a general Exception, but didn't see the annotation above where we check for the specific type. This is fine as is then.